meteor/server/api/playout/adlib.ts

```diff
@@ -402,22 +409,21 @@ export namespace ServerPlayoutAdLibAPI {
 		const span = profiler.startSpan('innerStartQueuedAdLib')
 		logger.info('adlibQueueInsertPartInstance')
 
-		// check if there's already a queued part after this:
-		const { nextPartInstance } = getSelectedPartInstancesFromCache(cache, rundownPlaylist)
-		if (nextPartInstance && nextPartInstance.part.dynamicallyInsertedAfterPartId) {
-			// TODO-PartInstance - pending new data flow - the call to setNextPart will prune the partInstance, so this will not be needed
-			cache.Parts.remove(nextPartInstance.part._id)
-			cache.PartInstances.remove(nextPartInstance._id)
-			cache.PieceInstances.remove({ partInstanceId: nextPartInstance._id })
-			afterRemoveParts(cache, currentPartInstance.rundownId, [nextPartInstance.part])
-		}
-
 		// Ensure it is labelled as dynamic
-		newPartInstance.part.dynamicallyInsertedAfterPartId = currentPartInstance.part._id
+		newPartInstance.orphaned = 'adlib-part'
+
+		const followingPart = selectNextPart(
+			rundownPlaylist,
+			currentPartInstance,
+			getAllOrderedPartsFromCache(cache, rundownPlaylist),
+			true
+		)
+		newPartInstance.part._rank = getRank(
+			currentPartInstance.part,
+			followingPart?.part?.segmentId === newPartInstance.segmentId ? followingPart?.part : undefined
+		)
 
 		cache.PartInstances.insert(newPartInstance)
-		// TODO-PartInstance - pending new data flow
-		cache.Parts.insert(newPartInstance.part)
```

Comment:
New data flow? Just checking this TODO does not need to be done?

Reply:
The 'new data flow' it was referring to was this PR :)